packages/remix-server-runtime/server.ts

```diff
@@ -147,6 +147,9 @@ async function handleDataRequest({
       let headers = new Headers(response.headers);
       headers.set("X-Remix-Redirect", headers.get("Location")!);
       headers.delete("Location");
+      if (response.headers.get("Set-Cookie") !== null) {
```

Comment:
In the browser, you can not access `response.headers.get("Set-Cookie")` to determine if a cookie was set in the response when `credentials: "same-origin"` is set. To avoid having to set it to `credentials: "include"` and adding additional CORS headers, this is my workaround.

Reply:
Seems fine to me, maybe X-Remix-Revalidate instead incase we discover other reasons to revalidate we can use the same mechanism?